### DISCUSSION TITLE
Mocreo ST4 temperature sensor - interview failure, random misdetection

### DISCUSSION THREAD

**Discussion by srd424** [Koenkk/zigbee2mqtt#13273] [Devices]
I'm new to z2m, so hope it's OK to post here? I have a "[Mocreo ST-4 temperature sensor](https://www.amazon.co.uk/MOCREO-Temperature-Waterproof-Thermometer-Refrigerator/dp/B09G9MG19D/)", which behaves rather oddly - it reports no manufacturer or model, only the string `mocreo.sensor_t` in the `dateCode` field. I've managed to get it working well enough with the following definition:
```
const definition = {
            fingerprint: [
                {dateCode: 'mocreo.sensor_t'},
            ],
            model: 'ST-4',
            vendor: 'Mocreo',
            description: 'Mocreo ST-4 temperature sensor (external probe)',
            fromZigbee: [fz.temperature], 
            toZigbee: [],
            exposes: [e.temperature()],
};
```
It generally fails to interview, but emits sensor readings regardless, which I can live with. However, I just had to re-pair it, and it was bizarrely misidentified as "Livolo Zigbee switch (1, 2, 3, 4 gang) (TI0001)" This may have been something to do with whether the external temperature probe was connected or not, possibly forcing it into a different state, or changes to the probe temperature keeping the device awake?

At some point it has been seen to emit the following as well:
```
Zigbee2MQTT:debug 2022-07-24 16:42:48: No converter available for 'ST-4' with cluster 'manuSpecificUbisysDimmerSetup' and type 'raw' and data '{"data":[24,10,10,6,0,0,0],"type":"Buffer"}'
```
It's designed to talk to a proprietary hub and I wonder if it's expecting a non-standard handshake or configuration command after pairing completes. I wondered if anyone had encountered any devices which behave similarly? I've looked at the code for the Livolo TI0001 and I can't see why it would have been misidentified. I know I should turn on herdsman debug logging and play around with re-pairing it, but now I've got it working I'm loathe to reset the device again...

**Comment by Koenkk** [comment]
Could you provide the herdsman debug log when pairing this device?

See https://www.zigbee2mqtt.io/guide/usage/debug.html on how to enable the herdsman debug logging. Note that this is only logged to STDOUT and not to log files.

**Comment by srd424** [comment]
OK, attached is an example with the external probe disconnected, when it simply fails to interview. I will try to provoke it into doing the weird misidentification thing in the next few days..
[interview-fail.txt](https://github.com/Koenkk/zigbee2mqtt/files/9184665/interview-fail.txt)

**Reply by Koenkk** [reply]
Can you provide the logging again but then:
1. force remove device through z2m frontend
2. restart z2m
3. start z2m, pair device (interview fails) <- provide the logging of this.

**Comment by srd424** [comment]
I moved the device to a fresh install of z2m and captured the logs, hopefully that is equivalent? There may be a bit of clutter in this one, because it's my live network now and there are obviously other devices. I did also try to wake the device during the interview attempt by pressing its button briefly, to no noticeable effect. 
[moncreo-fresh-interview-failed.txt](https://github.com/Koenkk/zigbee2mqtt/files/9226874/moncreo-fresh-interview-failed.txt)

**Comment by srd424** [comment]
Hmmm, the misdetection problem just happened as well, but after I'd stopped working on this and gone to do something else (defrost the freezer, ironically!) The logs may therefore be long and cluttered...
[mocreo-misdetect.txt](https://github.com/Koenkk/zigbee2mqtt/files/9226958/mocreo-misdetect.txt)

**Reply by Koenkk** [reply]
I indeed see the mis detect now, can you provide me the data/database.db entry of this device after it has been identified as the Livolo device?

**Comment by srd424** [comment]
I will do when I can face messing around with that sensor again - it always takes a while to get it back into a usable state afterwards!

TBF, it does it seem from logs that it did report "TI001" as a model number at some point, though without a manufacturer name. I'm beginning to think this device's firmware was just hacked together enough to work with the manufacturer's custom hub, I suspect it's a lost cause trying to get it working any further, but I'll bow to your experience there :) (Thanks for all the work on z2m by the way - I've just switched from deconz, and I'm really enjoying the extra flexibility, having a proper web admin interface, etc.!)

**Comment by srd424** [comment]
[mocreo.mis-id.raw.json.txt](https://github.com/Koenkk/zigbee2mqtt/files/9275329/mocreo.mis-id.raw.json.txt)
Finally managed to get it (raw version attached):
```
{
   "dateCode" : "mocreo.sensor_t",
   "defaultSendRequestWhen" : "immediate",
   "endpoints" : {
      "1" : {
         "binds" : [],
         "clusters" : {
            "genBasic" : {
               "attributes" : {
                  "dateCode" : "mocreo.sensor_t"
               }
            },
            "msTemperatureMeasurement" : {
               "attributes" : {
                  "measuredValue" : 2400
               }
            }
         },
         "configuredReportings" : [],
         "devId" : 255,
         "epId" : 1,
         "inClusterList" : [
            65261,
            1026,
            1029,
            64512,
            0,
            64513
         ],
         "meta" : {},
         "outClusterList" : [],
         "profId" : 260
      },
      "8" : {
         "binds" : [],
         "clusters" : {
            "genBasic" : {
               "attributes" : {
                  "hwVersion" : 1,
                  "manufacturerName" : "TexasInstruments",
                  "modelId" : "TI0001          ",
                  "powerSource" : 1,
                  "zclVersion" : 1
               }
            }
         },
         "configuredReportings" : [],
         "devId" : 770,
         "epId" : 8,
         "inClusterList" : [
            0,
            3,
            1026
         ],
         "meta" : {},
         "outClusterList" : [
            3
         ],
         "profId" : 260
      }
   },
   "epList" : [
      1,
      8
   ],
   "hwVersion" : 1,
   "id" : 21,
   "ieeeAddr" : "0x00124b0027277e31",
   "interviewCompleted" : true,
   "lastSeen" : 1659799075125,
   "manufId" : 0,
   "manufName" : "TexasInstruments",
   "meta" : {
      "configured" : 1
   },
   "modelId" : "TI0001          ",
   "nwkAddr" : 40967,
   "powerSource" : "Mains (single phase)",
   "type" : "EndDevice",
   "zclVersion" : 1
}
```

**Reply by Koenkk** [reply]
I found that there are more devices which use the `TI0001` zigbee model. Given that the manufacturer name is texas instruments (aka TI) I believe they didn't change the Zigbee model of the standard Texas Instruments examples.

The following fingerprint should do it:

```
        fingerprint: [
            {
                type: 'Router', manufacturerName: 'TexasInstruments', modelID: 'TI0001          ', endpoints: [
                    {ID: 1, profileID: 260, deviceID: 255, inputClusters: [65261,1026,1029,64512,0,64513], outputClusters: []},
                    {ID: 8, profileID: 260, deviceID: 770, inputClusters: [0,3,1026], outputClusters: [3]},
                ],
            },
        ],
```

**Reply by srd424** [reply]
Thanks - is it possible to have multiple fingerprints? Because how this device appears is very variable depending on how much it wakes up during the (attempted) interview process!

(I have no idea how the official hub that goes with this thing manages to recognise it - maybe just by MAC address prefix?)

**Reply by Koenkk** [reply]
Yes multiple fingerprints are possible
